# Louisville Free Public Library

Analysis of Louisville Free Public Library collection inventory.

Data Source: [Louisville Metro Open Data](https://data.louisvilleky.gov/datasets/LOJIC::louisville-metro-ky-library-collection-inventory-/about)

LFPL's collection inventory. Updated on a monthly basis.

## Definitions


- **BibNum** - The unique identifier of a bibliographic record within our materials database. Materials with the same bibliographic # will generally have the same cataloging metadata, differing only in the barcode number, assigned location and anything else specific to the individual copy.
- **Title**- The name of the material.
- **Author**- The writer or creator of the material.
- **ISBN**- The International Standard Book Number is a numeric commercial book identifier that is intended to be unique. Publishers purchase ISBNs from an affiliate of the International ISBN Agency. An ISBN is assigned to each separate edition and variation of a publication. 
- **Publication Year**- The year that the material was originally published.
- **Item Type**- Describes the type of material of each item, including Books, Audiobooks, Serials, DVDs, Microforms, Three Dimensional Objects, Kits, and Printed Cartographic Materials. 
- **Item Collection**- Refers to the collection the material belongs to based on common themes, including but not limited to Adult Fiction, Adult Reference, Mystery, Children’s Fiction, etc.  
- **Item Location**- The library location where the material was assigned at the time the report was run.
- **Item Price** – The price, in USD, that LFPL purchased the material for. 

## Exploring the Dataset

In [4]:
import pandas as pd
from pathlib import Path

data_path = Path('data/books.csv.gz')
books_df = pd.read_csv(data_path)
books_df.shape

(1190176, 10)

In [5]:
books_df.columns

Index(['BibNum', 'Title', 'Author', 'ISBN', 'PublicationYear', 'ItemType',
       'ItemCollection', 'ItemLocation', 'ItemPrice', 'ReportDate'],
      dtype='object')

In [6]:
books_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1190176 entries, 0 to 1190175
Data columns (total 10 columns):
 #   Column           Non-Null Count    Dtype  
---  ------           --------------    -----  
 0   BibNum           1190176 non-null  int64  
 1   Title            1190175 non-null  object 
 2   Author           1124225 non-null  object 
 3   ISBN             1153891 non-null  float64
 4   PublicationYear  1190176 non-null  int64  
 5   ItemType         1190176 non-null  object 
 6   ItemCollection   1190036 non-null  object 
 7   ItemLocation     1190176 non-null  object 
 8   ItemPrice        1190176 non-null  float64
 9   ReportDate       1190176 non-null  object 
dtypes: float64(2), int64(2), object(6)
memory usage: 90.8+ MB


In [7]:
books_df.isnull().sum().sort_values(ascending=False)

Author             65951
ISBN               36285
ItemCollection       140
Title                  1
BibNum                 0
PublicationYear        0
ItemType               0
ItemLocation           0
ItemPrice              0
ReportDate             0
dtype: int64

In [8]:
books_df

BibNum                                              Title  \
0         707409  Jeff Immelt and the new GE way : innovation, t...   
1         707411  Robin rescues dinner : 52 weeks of quick-fix m...   
2         707411  Robin rescues dinner : 52 weeks of quick-fix m...   
3         707411  Robin rescues dinner : 52 weeks of quick-fix m...   
4         707411  Robin rescues dinner : 52 weeks of quick-fix m...   
...          ...                                                ...   
1190171  2608597  25 ready-to-use sustainable living programs fo...   
1190172  2608598  Crypto basics : a nontechnical introduction to...   
1190173  2608598  Crypto basics : a nontechnical introduction to...   
1190174  2608599                                    Data governance   
1190175  2608599                                    Data governance   

                       Author          ISBN  PublicationYear ItemType  \
0         Magee, David, 1965-  9.780072e+12             2009     Book   
1        Miller, Robin, 1964-  9.780307e+12             2009     Book   
2        Miller, Robin, 1964-  9.780307e+12             2009     Book   
3        Miller, Robin, 1964-  9.780307e+12             2009     Book   
4        Miller, Robin, 1964-  9.780307e+12             2009     Book   
...                       ...           ...              ...      ...   
1190171                   NaN  9.780839e+12             2022     Book   
1190172         Gomzin, Slava  9.781484e+12             2022     Book   
1190173         Gomzin, Slava  9.781484e+12             2022     Book   
1190174  Reichental, Jonathan  9.781120e+12             2023     Book   
1190175  Reichental, Jonathan  9.781120e+12             2023     Book   

            ItemCollection            ItemLocation  ItemPrice  \
0        Adult Non-Fiction                    Main      25.95   
1        Adult Non-Fiction               Southwest      19.99   
2        Adult Non-Fiction               Southwest      19.99   
3        Adult Non-Fiction  Remote Shelving - Main      19.99   
4        Adult Non-Fiction  Remote Shelving - Main      19.99   
...                    ...                     ...        ...   
1190171  Adult Non-Fiction           South Central      63.69   
1190172  Adult Non-Fiction                 Bon Air      30.09   
1190173  Adult Non-Fiction                 Newburg      30.09   
1190174  Adult Non-Fiction                    Main      24.34   
1190175  Adult Non-Fiction             St Matthews      24.34   

                  ReportDate  
0        02/01/2023 00:00:00  
1        02/01/2023 00:00:00  
2        02/01/2023 00:00:00  
3        02/01/2023 00:00:00  
4        02/01/2023 00:00:00  
...                      ...  
1190171  02/01/2023 00:00:00  
1190172  02/01/2023 00:00:00  
1190173  02/01/2023 00:00:00  
1190174  02/01/2023 00:00:00  
1190175  02/01/2023 00:00:00  

[1190176 rows x 10 columns]

## Exploring the Columns

The columns of interest for this analysis include:

- BibNum
- Author
- Title
- PublicationYear
- ItemCollection
- ItemLocation
- ItemPrice


The following columns will be removed from the clean data:
- ISBN
- ReportDate

### Author

The author field appears to optionally contain dates. This is probably to 
help distinguish between two authors with the same name. No cleaning necessary
for the purposes of our analysis.

In [9]:
books_df['Author'].value_counts()

Patterson, James, 1947-                 5856
Osborne, Mary Pope.                     2063
Steel, Danielle                         1824
Pilkey, Dav, 1966-                      1812
Seuss, Dr.                              1812
                                        ... 
Sánchez Ferlosio, Rafael, 1927-2019       1
Pippen, Kitty, 1919-2018                   1
Adkins, Frank (Francis A.)                 1
Ray, James A.                              1
Kniffke, Sophie.                           1
Name: Author, Length: 187472, dtype: int64

### PublicationYear

The PublicationYear field contains years stored in number format ranging from 
0 to 9999. Most of the books in the collection have a publication year falling 
between 2005 and 2018.

**Cleaning Summary**

- 2971 records have a PublicationYear = 0. We will remove these records.
- 2 records have a PublicationYear = 9999. We will remove these records.
- 4 records have a PublicationYear = 2109. This is probably a typo. We will 
update these records to PublicationYear = 2019.

In [10]:
books_df['PublicationYear'].value_counts()

2018    77168
2017    72501
2015    70429
2016    70174
2020    68074
        ...  
1818        1
1794        1
1814        1
1807        1
1829        1
Name: PublicationYear, Length: 213, dtype: int64

In [11]:
books_df['PublicationYear'].describe().apply(lambda x: format(int(x), 'd'))

count    1190176
mean        2004
std          101
min            0
25%         2005
50%         2014
75%         2018
max         9999
Name: PublicationYear, dtype: object

In [12]:
# analysis courtesy of Jamie Smithers
bins = [1700, 1800, 1900, 2000, 2100]
labels = ['1700-1799', '1800-1899', '1900-1999', '2000-Present']
books_df['PublicationCategory'] = pd.cut(books_df['PublicationYear'], bins=bins, labels=labels, right=False)
books_df['PublicationCategory'].value_counts().apply(lambda x: format(int(x), 'n'))

2000-Present    1013675
1900-1999        172198
1800-1899          1323
1700-1799             3
Name: PublicationCategory, dtype: object

In [13]:
books_df[books_df['PublicationYear'] < 1700]

BibNum                                              Title  \
4439     1348353  Stolen legacy: the Egyptian origins of western...   
5660     1353780                                     LANKIE MANTITA   
6401     1362079                                         Act of War   
9405     1375916                                             Laptop   
9406     1375916                                             Laptop   
...          ...                                                ...   
1189832  2632061                  Black history month resource book   
1189833  2632062                                      The Bronx Zoo   
1189834  2632063                    Chas Addams half-baked cookbook   
1189835  2632064                            Shadow of the mountains   
1189836  2632065                                   Spring will come   

                         Author  ISBN  PublicationYear ItemType  \
4439         James, George G. M   NaN                0     Book   
5660          LESLIE PATRICELLI   NaN                0     Book   
6401                Brown, Dale   NaN                0     Book   
9405                        NaN   NaN                0     Book   
9406                        NaN   NaN                0     Book   
...                         ...   ...              ...      ...   
1189832  Snoodgrass, Mary Ellen   NaN                0     Book   
1189833            Lyle, Sparky   NaN                0     Book   
1189834         Addams, Charles   NaN                0     Book   
1189835            Morris, Lynn   NaN                0     Book   
1189836              Dye, Ginny   NaN                0     Book   

            ItemCollection   ItemLocation  ItemPrice           ReportDate  \
4439     Interlibrary Loan           Main        0.0  02/01/2023 00:00:00   
5660                   NaN        Shively        0.0  02/01/2023 00:00:00   
6401                   NaN        Bon Air        0.0  02/01/2023 00:00:00   
9405                Laptop  South Central     1077.0  02/01/2023 00:00:00   
9406                Laptop  South Central     1077.0  02/01/2023 00:00:00   
...                    ...            ...        ...                  ...   
1189832  Interlibrary Loan           Main        0.0  02/01/2023 00:00:00   
1189833  Interlibrary Loan           Main        0.0  02/01/2023 00:00:00   
1189834  Interlibrary Loan           Main        0.0  02/01/2023 00:00:00   
1189835  Interlibrary Loan           Main        0.0  02/01/2023 00:00:00   
1189836  Interlibrary Loan           Main        0.0  02/01/2023 00:00:00   

        PublicationCategory  
4439                    NaN  
5660                    NaN  
6401                    NaN  
9405                    NaN  
9406                    NaN  
...                     ...  
1189832                 NaN  
1189833                 NaN  
1189834                 NaN  
1189835                 NaN  
1189836                 NaN  

[2971 rows x 11 columns]

In [14]:
books_df[books_df['PublicationYear'] > 2023]

BibNum                                              Title  \
238815    407990  Residual life assessment of major light water ...   
532987    820258       The Greene murder case : a Philo Vance story   
1124391  2230395                    Murder knocks twice : a mystery   
1124392  2230395                    Murder knocks twice : a mystery   
1124393  2230395                    Murder knocks twice : a mystery   
1124394  2230395                    Murder knocks twice : a mystery   

                   Author          ISBN  PublicationYear ItemType  \
238815                NaN           NaN             9999     Book   
532987    Van Dine, S. S.           NaN             9999     Book   
1124391  Calkins, Susanna  9.781250e+12             2109     Book   
1124392  Calkins, Susanna  9.781250e+12             2109     Book   
1124393  Calkins, Susanna  9.781250e+12             2109     Book   
1124394  Calkins, Susanna  9.781250e+12             2109     Book   

               ItemCollection   ItemLocation  ItemPrice           ReportDate  \
238815   Government Documents           Main      75.00  02/01/2023 00:00:00   
532987                Mystery           Main      14.95  02/01/2023 00:00:00   
1124391               Mystery  South Central      10.49  02/01/2023 00:00:00   
1124392               Mystery      Northeast      10.49  02/01/2023 00:00:00   
1124393               Mystery  Jeffersontown      10.49  02/01/2023 00:00:00   
1124394               Mystery     Middletown      16.19  02/01/2023 00:00:00   

        PublicationCategory  
238815                  NaN  
532987                  NaN  
1124391                 NaN  
1124392                 NaN  
1124393                 NaN  
1124394                 NaN

### ItemCollection

The ItemCollection field contains multiple pieces of information including the 
book's genre, format, and intended audience. This field will be central to our 
analysis. 

**Cleaning Summary**
We will create 2 separate fields based on ItemCollection: 

- Genre:
    - **Fiction**: ['Adult Fiction', 'Mystery',
       'Older Teen Fiction', 'Younger Teen  Fiction',
       'Science Fiction', "Children's Fiction", 'Western',
       'Urban Fiction', 'Bestsellers', 'Storytime Collection',
       'Caldecott/Newbery']
    - **Non-Fiction**: ['Adult Non-Fiction','Teen Non-Fiction', "Children's Non-Fiction",
       'Natural Resources', 'Kentucky History','Adult Reference','Caldecott/Newbery','Government Documents',
       'Telereference',"Children's Reference", 'Teen Reference',  "Children's Non-Fiction Paperback"]
    - **Unknown**: ['Adult Paperback',
       "Children's Picture Paperback", "Children's Paperback",
       "Children's Picture Book", 'International Collection',
       'ELL Collection', 'Holiday', 'Oversize','Bestsellers', 'Storytime Collection',
       "Children's Board Book", "Children's Easy Reader",
       'Preschool  Picture Book', 'Interlibrary Loan',
       'Adult Paperbacks Tall', "Children's Easy Reader Paperback",
       'Laptop', 'Large Print', 'Telereference',
       'Big Book', 'College Shop', 'Magazines and Newspaper',
       'Younger Teen  Paperback', 'Younger Teen  Audiobook', 'Readalongs',
       'Listening Device', 'Adult Audiobook', 'Electronic',
       "Children's Music", 'Adult DVD', 'Music', 'Book Discussion Kit']
- Audience:
    - **Adult**: ['Adult Non-Fiction', 'Adult Fiction', 'Adult Paperback',
       'Adult Reference','Adult Paperbacks Tall','Adult Audiobook','Adult DVD']
    - **Teen**: ['Older Teen Fiction', 'Younger Teen  Fiction', 'Teen Non-Fiction', 'Teen Reference',
       'Younger Teen  Paperback', 'Younger Teen  Audiobook']
    - **Children**: ["Children's Fiction","Children's Picture Paperback", "Children's Paperback",
       "Children's Picture Book", "Children's Non-Fiction", 'Readalongs',
       "Children's Board Book", "Children's Easy Reader", 'Storytime Collection',
       'Preschool  Picture Book',"Children's Easy Reader Paperback",
       "Children's Non-Fiction Paperback", "Children's Reference", 'Teen Reference',
       "Children's Music"]
    - **Unknown**: ['Mystery', 'Science Fiction', 'Western','International Collection',
       'ELL Collection', 'Holiday', 'Natural Resources', 'Kentucky History', 'Oversize',
       'Urban Fiction', 'Bestsellers', 'Interlibrary Loan',
       nan, 'Caldecott/Newbery', 'Laptop', 'Government Documents',
       'Large Print', 'Telereference', 'Big Book', 'College Shop', 'Magazines and Newspaper',       
       'Listening Device', 'Electronic','Music', 'Book Discussion Kit']

Once created, we can use these columns to identify YA books as:
- Genre = Fiction
- Audience = Teen

Finally - there are 140 records with no value for ItemCollection. We will remove 
these from the data set.

In [15]:
books_df['ItemCollection'].value_counts()

Adult Non-Fiction                   371433
Adult Fiction                       177604
Children's Non-Fiction               86356
Mystery                              60314
Children's Picture Book              59348
Preschool  Picture Book              51276
Children's Fiction                   48446
Adult Paperback                      45302
Children's Paperback                 45076
Children's Easy Reader               24511
Teen Non-Fiction                     24376
Older Teen Fiction                   23787
Children's Board Book                20057
Younger Teen  Fiction                17532
Kentucky History                     16962
Science Fiction                      16048
Children's Easy Reader Paperback     15959
Holiday                              15583
International Collection             15581
Adult Reference                      11197
Children's Picture Paperback          9731
Urban Fiction                         7601
Caldecott/Newbery                     6777
Storytime C

In [16]:
books_df['ItemCollection'].unique()

array(['Adult Non-Fiction', 'Adult Fiction', 'Mystery',
       'Older Teen Fiction', 'Younger Teen  Fiction', 'Adult Paperback',
       'Science Fiction', "Children's Fiction", 'Western',
       "Children's Picture Paperback", "Children's Paperback",
       "Children's Picture Book", 'International Collection',
       'ELL Collection', 'Teen Non-Fiction', "Children's Non-Fiction",
       'Holiday', 'Natural Resources', 'Kentucky History', 'Oversize',
       'Urban Fiction', 'Bestsellers', 'Storytime Collection',
       "Children's Board Book", "Children's Easy Reader",
       'Preschool  Picture Book', 'Adult Reference', 'Interlibrary Loan',
       nan, 'Adult Paperbacks Tall', "Children's Easy Reader Paperback",
       'Caldecott/Newbery', 'Laptop', 'Government Documents',
       'Large Print', 'Telereference', "Children's Non-Fiction Paperback",
       'Big Book', "Children's Reference", 'Teen Reference',
       'College Shop', 'Magazines and Newspaper',
       'Younger Teen  Paperba

In [17]:
books_df[books_df['ItemCollection'].isna()]


BibNum                                            Title  \
5660     1353780                                   LANKIE MANTITA   
6401     1362079                                       Act of War   
17032    1391304                                     Wagon Wheels   
17883    1392496                                  The Book of Liz   
18639    1393469  HAUNTED ENCOUNTERS: DEPARTED FAMILY AND FRIENDS   
...          ...                                              ...   
1185558  2608114                     Life Under the Jim Crow Laws   
1185755  2585123                                   THE SEX EFFECT   
1186251  2585283                                 The Way of Kings   
1188339  2626996                                  christmas smthn   
1189452  2632438                                       Odd Thomas   

                     Author  ISBN  PublicationYear ItemType ItemCollection  \
5660      LESLIE PATRICELLI   NaN                0     Book            NaN   
6401            Brown, Dale   NaN                0     Book            NaN   
17032       Barbara Brenner   NaN                0     Book            NaN   
17883               Sedaris   NaN                0     Book            NaN   
18639                   NaN   NaN                0     Book            NaN   
...                     ...   ...              ...      ...            ...   
1185558                 NaN   NaN                0     Book            NaN   
1185755         BENES, ROSS   NaN                0     Book            NaN   
1186251  Sanderson, Brandon   NaN                0     Book            NaN   
1188339                 NaN   NaN                0     Book            NaN   
1189452         Dean Koontz   NaN                0     Book            NaN   

                 ItemLocation  ItemPrice           ReportDate  \
5660                  Shively        0.0  02/01/2023 00:00:00   
6401                  Bon Air        0.0  02/01/2023 00:00:00   
17032                 Shively        0.0  02/01/2023 00:00:00   
17883                    Main        0.0  02/01/2023 00:00:00   
18639                 Bon Air        0.0  02/01/2023 00:00:00   
...                       ...        ...                  ...   
1185558             Southwest        0.0  02/01/2023 00:00:00   
1185755                  Main        0.0  02/01/2023 00:00:00   
1186251                  Main        0.0  02/01/2023 00:00:00   
1188339  Childrens Bookmobile        0.0  02/01/2023 00:00:00   
1189452         South Central        0.0  02/01/2023 00:00:00   

        PublicationCategory  
5660                    NaN  
6401                    NaN  
17032                   NaN  
17883                   NaN  
18639                   NaN  
...                     ...  
1185558                 NaN  
1185755                 NaN  
1186251                 NaN  
1188339                 NaN  
1189452                 NaN  

[140 rows x 11 columns]

### ItemLocation

**Cleaning Summary**

No cleaning needed.

In [18]:
books_df['ItemLocation'].value_counts()

Remote Shelving - Main       139987
Northeast                    124473
Southwest                    122113
Main                         121439
South Central                115837
Bon Air                       74730
St Matthews                   69531
Jeffersontown                 56706
Iroquois                      52382
Highlands - Shelby Park       45539
Crescent Hill                 42837
Childrens Main Library        38994
Middletown                    33120
Shively                       23623
Newburg                       23586
Fairdale                      23149
Shawnee                       22906
Western                       21648
Portland                      13334
Childrens Bookmobile           9129
Remote Shelving - Shawnee      9083
Main Teen                      6024
Content Management                4
Adult Bookmobile                  2
Name: ItemLocation, dtype: int64

### ItemPrice

The least expensive book cost $0 and the most expensive book cost $1,077. The
average book price is $18.

**Cleaning Summary**

No cleaning needed.

In [19]:
books_df['ItemPrice'].describe().apply(lambda x: format(int(x), 'f'))

count    1190176.000000
mean          18.000000
std           15.000000
min            0.000000
25%           10.000000
50%           15.000000
75%           24.000000
max         1077.000000
Name: ItemPrice, dtype: object

In [20]:
books_df[books_df['ItemPrice'] == 0]['ItemCollection'].value_counts()

Interlibrary Loan               162
Adult Non-Fiction                62
Government Documents             47
Kentucky History                 46
Adult Fiction                    16
Teen Non-Fiction                 10
Mystery                          10
Children's Non-Fiction            8
Children's Board Book             8
Storytime Collection              7
Preschool  Picture Book           7
Western                           7
Adult Paperback                   7
Children's Picture Book           6
Children's Paperback              5
Science Fiction                   5
International Collection          4
Younger Teen  Fiction             4
Caldecott/Newbery                 3
Children's Picture Paperback      3
Children's Fiction                3
Electronic                        3
Older Teen Fiction                3
Holiday                           2
Children's Easy Reader            2
Adult Reference                   1
Magazines and Newspaper           1
Name: ItemCollection, dtype:

## Cleaning Summary


1. Remove unneeded columns (ISBN, ReportDate)
1. Remove records with empty and invalid PuublicationYear or ItemCollection.
   - 2971 records have a PublicationYear = 0. We will remove these records.
   - 2 records have a PublicationYear = 9999. We will remove these records.
   - 140 records with no value for ItemCollection. We will remove these records.
1. Update incorrect values (PublicationYear 2109 -> 2019)
   - 4 records have a PublicationYear = 2109. This is probably a typo. We will update these records to PublicationYear = 2019.
1. Add genre and audience columns based on the value of ItemCollection:

Genre:
- **Fiction**: ['Adult Fiction', 'Mystery',
       'Older Teen Fiction', 'Younger Teen  Fiction',
       'Science Fiction', "Children's Fiction", 'Western',
       'Urban Fiction', 'Bestsellers', 'Storytime Collection']
- **Non-Fiction**: ['Adult Non-Fiction','Teen Non-Fiction', "Children's Non-Fiction",
       'Natural Resources', 'Kentucky History','Adult Reference','Government Documents',
       'Telereference',"Children's Reference", 'Teen Reference',  "Children's Non-Fiction Paperback"]
- **Unknown**: ['Adult Paperback',
       "Children's Picture Paperback", "Children's Paperback",
       "Children's Picture Book", 'International Collection','Caldecott/Newbery',
       'ELL Collection', 'Holiday', 'Oversize','Bestsellers', 'Storytime Collection',
       "Children's Board Book", "Children's Easy Reader",
       'Preschool  Picture Book', 'Interlibrary Loan',
       'Adult Paperbacks Tall', "Children's Easy Reader Paperback",
       'Laptop', 'Large Print', 'Telereference',
       'Big Book', 'College Shop', 'Magazines and Newspaper',
       'Younger Teen  Paperback', 'Younger Teen  Audiobook', 'Readalongs',
       'Listening Device', 'Adult Audiobook', 'Electronic',
       "Children's Music", 'Adult DVD', 'Music', 'Book Discussion Kit']

Audience:
- **Adult**: ['Adult Non-Fiction', 'Adult Fiction', 'Adult Paperback',
       'Adult Reference','Adult Paperbacks Tall','Adult Audiobook','Adult DVD']
- **Teen**: ['Older Teen Fiction', 'Younger Teen  Fiction', 'Teen Non-Fiction', 'Teen Reference',
       'Younger Teen  Paperback', 'Younger Teen  Audiobook']
- **Children**: ["Children's Fiction","Children's Picture Paperback", "Children's Paperback",
       "Children's Picture Book", "Children's Non-Fiction", 'Readalongs',
       "Children's Board Book", "Children's Easy Reader", 'Storytime Collection',
       'Preschool  Picture Book',"Children's Easy Reader Paperback",'Caldecott/Newbery',
       "Children's Non-Fiction Paperback", "Children's Reference", 'Teen Reference',
       "Children's Music"]
- **Unknown**: ['Mystery', 'Science Fiction', 'Western','International Collection',
       'ELL Collection', 'Holiday', 'Natural Resources', 'Kentucky History', 'Oversize',
       'Urban Fiction', 'Bestsellers', 'Interlibrary Loan',
       nan, 'Laptop', 'Government Documents',
       'Large Print', 'Telereference', 'Big Book', 'College Shop', 'Magazines and Newspaper',       
       'Listening Device', 'Electronic','Music', 'Book Discussion Kit']

Once created, we can use these columns to identify YA books as:
- Genre = Fiction
- Audience = Teen

